modules/riemann/riemann-worker.c

```diff
@@ -348,25 +348,41 @@ riemann_worker_flush(LogThreadedDestWorker *s, LogThreadedFlushMode mode)
                                                  MAX(1, owner->super.batch_lines));
   if (!r)
     {
+      msg_error("riemann: error calling riemann_communicate()",
+                evt_tag_str("server", owner->server),
+                evt_tag_int("port", owner->port),
+                evt_tag_int("batch_size", self->event.n),
+                evt_tag_int("ok", r->ok),
+                evt_tag_str("error", r->error),
```

Comment:
In this codepath, `r` is `NULL`, so we can't access `ok` / `error` here.

Reply:
thanks for the note, I've fixed this.